### PR TITLE
deposit function params and event update

### PR DESCRIPTION
**Pull request by psorinionut**

### CODE REVIEW

File: esdt-safe/src/to_sovereign/events.rs [hunk 1]
```diff
@@ -38,7 +36,7 @@ pub trait EventsModule {
     fn deposit_event(
         &self,
         #[indexed] dest_address: &ManagedAddress,
-        #[indexed] tokens: &PaymentsVec<Self::Api>,
+        #[indexed] tokens: &MultiValueEncoded<MultiValue3<TokenIdentifier, u64, BigUint>>,
```
Comment (dorin-iancu) [inline esdt-safe/src/to_sovereign/events.rs:39]: Don't use MultiValue types for events. It's bad practice, and doesn't do what you expect it to. Revert to PaymentsVec.

**Review by dorin-iancu — CHANGES_REQUESTED** [review]

File: esdt-safe/src/to_sovereign/events.rs (hunk 1 above)
Comment (sasurobert) [reply esdt-safe/src/to_sovereign/events.rs:39]: @dorin-iancu can you fix this, as you work in this repo ?

File: esdt-safe/src/to_sovereign/create_tx.rs [hunk 2]
```diff
@@ -14,29 +14,41 @@ pub trait CreateTxModule:
...
-    /// Create an Elrond -> Sovereign transaction.
+    /// Create an MultiversX -> Sovereign transaction.
     #[payable("*")]
     #[endpoint]
     fn deposit(
         &self,
         to: ManagedAddress,
-        opt_transfer_data: OptionalValue<TransferData<Self::Api>>,
+        opt_gas_limit: OptionalValue<GasLimit>,
+        opt_function: OptionalValue<ManagedBuffer>,
```
Comment (sasurobert) [inline esdt-safe/src/to_sovereign/create_tx.rs:24]: too many optionals - it is a bad practice. rewrite this. The thing is if we have gasLimit - we must have a function and we might have opt_args.

So make optional a structure which will have the GasLimit, Function and ArgsList.

**Review by sasurobert — CHANGES_REQUESTED** [review]

**Comment by sasurobert** [comment]
closed, as it was resolved in another PR.

**Pull request closed without merging**